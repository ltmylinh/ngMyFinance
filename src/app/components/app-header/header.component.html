
<mat-toolbar class="header__toolbar">
  <div class="header__toggle-button">
      <button mat-icon-button (click)="onToggleSideNav()"><mat-icon>menu</mat-icon></button>
  </div>
  <div class="header__right">
      <a class="header__right__logo">
          <img src="/assets/images/logo.png" alt="My Finance" />
      </a>
      <div class="header__right__search">
          <form *ngIf="toggleSearch" class="header__right__search__form">
            <button type="button" (click)="search()" mat-icon-button>
                <mat-icon>search</mat-icon>
            </button>
            <mat-form-field class="header__right__search__input">
                <input matInput placeholder="Search">
            </mat-form-field>
            <button type="button" (click)="toggle()" mat-icon-button>
                <mat-icon>close</mat-icon>
            </button>
          </form>
          <button (click)="toggle()" mat-button *ngIf="!toggleSearch">
              <mat-icon>search</mat-icon>
              Search
          </button>
          <a class="header__right__search_signin" mat-button routerLink="auth/login">Signin</a>
          <button mat-icon-button type="button" (click)="onLogout()">Logout</button>
        </div>
  </div>
</mat-toolbar>

