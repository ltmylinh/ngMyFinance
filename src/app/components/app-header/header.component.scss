.header{
  display: flex;
  flex-direction: column;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  &__toolbar{
    position: fixed;
    z-index: 2;
    background-color: #fff;
    color: #aaacc2;
    padding: 0;
    box-shadow: 0 0 2px rgba(0, 0, 0, 0.3);
  }
  &__toggle-button{
    width: 64px;
    height: 64px;
    background: #673ab7;
    color: rgba(255,255,255,.87);
    display: flex;
    .mat-icon-button{
      width: 100%;
      height: 100%;
    }
  }
  &__right{
    display: flex;
    flex-grow: 2;
    &__logo{
      img{
        max-width: 100px;
        max-height: 50px;
      }
    }
    &__search{
      flex-grow: 2;
      display: flex;
      justify-content: flex-end;
      align-items: center;
      color: #676c7b;
      &__form{
        max-height: 36px;
      }
      .auth-buttons{
        margin-left: 10px;
        margin-right: 10px;
      }
    }
    &__search__input.mat-form-field {
      font-size: 0.75rem;
      .mat-form-field-underline{
        background-color: #ffffde;
      }
    }
  }
}
