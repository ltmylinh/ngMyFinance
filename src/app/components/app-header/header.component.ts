import { Component, Output, EventEmitter } from '@angular/core';

@Component({
    selector: 'app-header',
    styleUrls: ['header.component.scss'],
    template: `
        <div class="header">
          <mat-toolbar color="primary" class="header__toolbar">
            <button mat-icon-button (click)="onToggleSideNav()"><mat-icon>menu</mat-icon></button>
            <h1 class="example-app-name">Responsive App</h1>
          </mat-toolbar>
        </div>
    `
})
export class AppHeaderComponent {
    @Output()
      toggleNav = new EventEmitter<any>();

    constructor() {}

    onToggleSideNav(){
      this.toggleNav.emit();
    }
}
