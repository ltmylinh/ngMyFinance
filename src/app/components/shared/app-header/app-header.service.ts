import { Injectable } from '@angular/core';

import { Store } from 'store';

import { Observable } from 'rxjs/Observable';
import { Subject } from 'rxjs/Subject';
import 'rxjs/add/operator/do';

@Injectable()
export class AppHeaderService {

  private click$ = new Subject();

  openNav$: Observable<any> = this.click$
    .do((next: boolean) => {
      this.store.set('isOpenNav', next);
    });

  constructor(
    private store: Store
  ) {}

  updateIsOpenNav(value: boolean){
    this.click$.next(value);
  }
}
