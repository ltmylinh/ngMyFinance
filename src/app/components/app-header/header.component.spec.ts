import {
  TestBed,
  ComponentFixture,
  async
} from '@angular/core/testing';

import { AppHeaderComponent } from './header.component';
import { AppHeaderService } from './../shared/app-header/app-header.service';
import { AngularMaterialModule } from './../../common-modules/angular-material.module';

describe('HeaderComponent', () => {
  let component: AppHeaderComponent;
  let appHeaderServiceStub = {};
  let appHeaderService: any;
  let fixture: ComponentFixture<AppHeaderComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [AngularMaterialModule],
      declarations: [AppHeaderComponent],
      providers: [{ provide: AppHeaderService, useValue: appHeaderServiceStub }]
    });

    appHeaderService = TestBed.get(AppHeaderService);
    fixture = TestBed.createComponent(AppHeaderComponent);
    component = fixture.componentInstance;
  });

  test('appHeaderService and appHeaderServiceStub are different', () => {

    expect(appHeaderService === appHeaderServiceStub).toBe(false);
  });

  test('test support snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

});
