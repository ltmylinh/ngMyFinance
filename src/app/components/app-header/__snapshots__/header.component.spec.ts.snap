// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent test support snapshot 1`] = `
<app-header
  headerService={[Function Object]}
  toggleSearch="false"
>
  
  <mat-toolbar
    class="header__toolbar mat-toolbar"
  >
    
    
    <div
      class="header__toggle-button"
    >
      
        
      <button
        class="mat-icon-button"
        mat-icon-button=""
      >
        <span
          class="mat-button-wrapper"
        >
          <mat-icon
            aria-hidden="true"
            class="mat-icon"
            role="img"
          >
            menu
          </mat-icon>
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
      
    
    </div>
    
    
    <div
      class="header__right"
    >
      
        
      <a
        class="header__right__logo"
      >
        
            
        <img
          alt="My Finance"
          src="/assets/images/logo.png"
        />
        
        
      </a>
      
        
      <div
        class="header__right__search"
      >
        
            
        
        
            
        
        
            
        
          
      </div>
      
    
    </div>
    
  
  </mat-toolbar>
  
  
</app-header>
`;
